File: frontend/src/api/crm.ts
import { fetchApi } from './client';

export interface CRMExportResponse {
  export_id: string;
  buyer_id: string;
  buyer_name: string;
  format: string;
  status: string;
  payload: Record<string, any>;
  download_url?: string;
  message: string;
}

export const exportCRM = (buyerId: string, format: 'hubspot' | 'salesforce' | 'csv') =>
  fetchApi<CRMExportResponse>('/api/v1/crm/export', {
    method: 'POST',
    body: JSON.stringify({ buyer_id: buyerId, export_format: format }),
  });

File: backend/app/schemas/crm.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any

class CRMExportRequest(BaseModel):
    buyer_id: str
    export_format: str = "hubspot"  # hubspot | salesforce | csv
    include_outreach_sequence: bool = True
    include_eudr_pack: bool = True
    webhook_url: Optional[str] = None

class CRMExportResponse(BaseModel):
    export_id: str
    buyer_id: str
    buyer_name: str
    format: str
    status: str
    payload: Dict[str, Any]
    download_url: Optional[str] = None
    message: str
